test/lit/passes/optimize-instructions-gc.wast

```diff
@@ -2841,8 +2839,11 @@
 
   ;; CHECK:      (func $refinalize.select.arm (type $void)
   ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (ref.cast (ref $void2)
```

Comment:
@kripken, do you know offhand what the purpose of this cast is in this test? Is it ok that it's now optimized out? Should we remove it from the input? Same for the next couple of tests.